spinn_utilities/executable_finder.py

```diff
@@ -56,20 +58,22 @@ def get_executable_path(self, executable_name):
                 return potential_filename
 
         # No executable found
-        return None
+        raise FailedToFindBinaryException(
```

Comment:
Why are you introducing a new failure mode?

Reply:
better to fail before loading data. if you dont have a failure mode here. it causes some horrible error about flood filling None. fail fast and understandablly